scalafmt-core/shared/src/main/scala/org/scalafmt/util/OsSpecific.scala

```diff
@@ -3,8 +3,7 @@ package org.scalafmt.util
 /** Utils related to differences between various operating systems. */
 object OsSpecific {
   val isWindows: Boolean =
-    System.getProperty("os.name").toLowerCase().startsWith("windows")
-  val lineSeparator: String = System.getProperty("line.separator")
+    System.getProperty("os.name", "unknown").toLowerCase().startsWith("windows")
```

Comment:
What about this check when running on Node.js? It's still relevant, but then `System.getProperty("os.name")` won't work. 

This works on Node (probably there's a simpler way):

```scala
scalajs.js.Dynamic.global.require("os").platform().startsWith("win")
``` 

But won't work when running in a browser. 

Anyway it requires having this piece of code split for JVM and JS.

Reply:
How about `def isWindows = System.lineSeparator == "\r\n"`?

Reply:
org.scalameta:io actually implements `java.io.File.separator` https://github.com/scalameta/scalameta/blob/5536eb02db562af944f80c11cb171e19508771c4/scalameta/io/js/src/main/scala/java/io/File.scala#L57-L62 we can use that instead, it does the right thing on node.js and assumes `/` in the browser

Reply:
Good idea! Seems to work both on Node and in the browser: returns `\n` on Mac, hopefully works on Windows as expected.

Reply:
Sure, if it's already done in scalameta, why not. Then it probably should be used below in `fixSeparatorsInPathPattern` too

Reply:
Done!